extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageUtil.java

```diff
@@ -388,12 +392,65 @@ private static ConvertedArrowSchema convertArrowSchema(
         result.tableDef = TableDefinition.of(columns);
 
         result.attributes = new HashMap<>();
+
+        final HashMap<String, String> attributeTypeMap = new HashMap<>();
         tableMetadataVisitor.accept((key, value) -> {
-            final String prefix = ATTR_DH_PREFIX + ATTR_ATTR_TAG + ".";
-            if (!key.startsWith(prefix)) {
+            final String isAttributePrefix = ATTR_DH_PREFIX + ATTR_ATTR_TAG + ".";
+            final String isAttributeTypePrefix = ATTR_DH_PREFIX + ATTR_ATTR_TYPE_TAG + ".";
+            if (key.startsWith(isAttributePrefix)) {
+                result.attributes.put(key.substring(isAttributePrefix.length()), value);
+            } else if (key.startsWith(isAttributeTypePrefix)) {
+                attributeTypeMap.put(key.substring(isAttributeTypePrefix.length()), value);
```

Comment:
looks unrelated, would you consider breaking this out and/or also including this in the js impl, so that it can have support for knowing what the types of attrs should be?